### PR TITLE
add testsuite to invoke the makedata suite from the RTA

### PR DESCRIPTION
**Pull request by dothebart**
### Scope & Purpose

RTA has a set of javascript tests that check the integrity of certain structures. 
### Checklist

- [x] this is a testsuite.
- [x] Backports
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*
  - [ ] Backport for 3.7: *(Please link PR)*

### CODE REVIEW

**Review by adityamukho — APPROVED** [review]
Just one minor suggestion. Otherwise LGTM. Works locally.

**Review by adityamukho — COMMENTED** [review]
Just one question (in comments). Otherwise runs fine locally.

File: js/client/modules/@arangodb/testutils/process-utils.js [hunk 1]
```diff
@@ -1558,6 +1560,80 @@ function shutdownArangod (arangod, options, forceTerminate) {
...
 }
 
+
+function shutDownOneInstance(options, arangod, fullInstance, counters, forceTerminate, timeout) {
+  let shutdownTime = internal.time();
+  if (!arangod.hasOwnProperty('exitStatus')) {
+    if ((counters.nonAgenciesCount > 0) && (arangod.role === 'agent')) {
+      return true;
+    }
+    shutdownArangod(arangod, options, forceTerminate);
```
[the remark this answers is not in the document]
Comment (dothebart) [reply js/client/modules/@arangodb/testutils/process-utils.js:1570]: hm, you're onto something here, this should shut down any instance, agencies as well. Will remove that part.

**Comment by dothebart** [comment]
ARM compile job failed because of CI issues. This PR doesn't contain c++ changes.

File: js/client/modules/@arangodb/testsuites/rta_makedata.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  let rc = tu.performTests(localOptions, ['rta'], 'rta_makedata_test', runRtaInArangosh);
+  options.cleanup = options.cleanup && localOptions.cleanup;
+  return rc;
+}
+
+
+exports.setup = function (testFns, defaultFns, opts, fnDocs, optionsDoc, allTestPaths) {
+  Object.assign(allTestPaths, testPaths);
+  testFns['rta_makedata'] = makeDataWrapper;
+  opts['rtasource'] = fs.makeAbsolute(fs.join('.', '..','release-test-automation'));
```
Comment (mchacki) [inline js/client/modules/@arangodb/testsuites/rta_makedata.js:142]: Is this now hardCoded to `./../release-test-automation`?
Can we make it configureable from the outside please, on my system I do not have the RTA at exactly this position

Comment (dothebart) [reply js/client/modules/@arangodb/testsuites/rta_makedata.js:142]: this is the way, the testing.js modules interface accepts sensible defaults.
the array altered by this is the default values array: https://github.com/arangodb/arangodb/blob/devel/js/client/modules/%40arangodb/testutils/testing.js#L164 
which is preset to the actual values here: https://github.com/arangodb/arangodb/blob/devel/js/client/modules/%40arangodb/testutils/testing.js#L588 
and can be overridden from the CLI by specifying `--rtasource somefolder`

**Review by mchacki — APPROVED** [review]
LGTM.
Should not be included into Jenkins / Oskar / all target as it is a developer only tool and depends on the "third-party" RTA repository.

**Pull request merged**